File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/DomainSearchResponse.cpp


#include "DomainSearchResponse.h"

using namespace Tiny;

DomainSearchResponse::DomainSearchResponse()
{
	success = bool(false);
	response_text = std::string();
	response_time = std::string();
	lookup = std::list<Object>();
	suggest = std::list<Object>();
	tlds = std::list<std::string>();
}

DomainSearchResponse::DomainSearchResponse(std::string jsonString)
{
	this->fromJson(jsonString);
}

DomainSearchResponse::~DomainSearchResponse()
{

}

void
DomainSearchResponse::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *successKey = "success";

    if(object.has_key(successKey))
    {
        bourne::json value = object[successKey];



        jsonToValue(&success, value, "bool");


    }

    const char *response_textKey = "response_text";

    if(object.has_key(response_textKey))
    {
        bourne::json value = object[response_textKey];



        jsonToValue(&response_text, value, "std::string");


    }

    const char *response_timeKey = "response_time";

    if(object.has_key(response_timeKey))
    {
        bourne::json value = object[response_timeKey];



        jsonToValue(&response_time, value, "std::string");


    }

    const char *lookupKey = "lookup";

    if(object.has_key(lookupKey))
    {
        bourne::json value = object[lookupKey];


        std::list<Object> lookup_list;
        Object element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            lookup_list.push_back(element);
        }
        lookup = lookup_list;


    }

    const char *suggestKey = "suggest";

    if(object.has_key(suggestKey))
    {
        bourne::json value = object[suggestKey];


        std::list<Object> suggest_list;
        Object element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            suggest_list.push_back(element);
        }
        suggest = suggest_list;


    }

    const char *tldsKey = "tlds";

    if(object.has_key(tldsKey))
    {
        bourne::json value = object[tldsKey];


        std::list<std::string> tlds_list;
        std::string element;
        for(auto& var : value.array_range())
        {

            jsonToValue(&element, var, "std::string");


            tlds_list.push_back(element);
        }
        tlds = tlds_list;


    }


}

bourne::json
DomainSearchResponse::toJson()
{
    bourne::json object = bourne::json::object();





    object["success"] = isSuccess();






    object["response_text"] = getResponseText();






    object["response_time"] = getResponseTime();





    std::list<Object> lookup_list = getLookup();
    bourne::json lookup_arr = bourne::json::array();

    for(auto& var : lookup_list)
    {
        Object obj = var;
        lookup_arr.append(obj.toJson());
    }
    object["lookup"] = lookup_arr;






    std::list<Object> suggest_list = getSuggest();
    bourne::json suggest_arr = bourne::json::array();

    for(auto& var : suggest_list)
    {
        Object obj = var;
        suggest_arr.append(obj.toJson());
    }
    object["suggest"] = suggest_arr;






    std::list<std::string> tlds_list = getTlds();
    bourne::json tlds_arr = bourne::json::array();

    for(auto& var : tlds_list)
    {
        tlds_arr.append(var);
    }
    object["tlds"] = tlds_arr;






    return object;

}

bool
DomainSearchResponse::isSuccess()
{
	return success;
}

void
DomainSearchResponse::setSuccess(bool success)
{
	this->success = success;
}

std::string
DomainSearchResponse::getResponseText()
{
	return response_text;
}

void
DomainSearchResponse::setResponseText(std::string response_text)
{
	this->response_text = response_text;
}

std::string
DomainSearchResponse::getResponseTime()
{
	return response_time;
}

void
DomainSearchResponse::setResponseTime(std::string response_time)
{
	this->response_time = response_time;
}

std::list<Object>
DomainSearchResponse::getLookup()
{
	return lookup;
}

void
DomainSearchResponse::setLookup(std::list<Object> lookup)
{
	this->lookup = lookup;
}

std::list<Object>
DomainSearchResponse::getSuggest()
{
	return suggest;
}

void
DomainSearchResponse::setSuggest(std::list<Object> suggest)
{
	this->suggest = suggest;
}

std::list<std::string>
DomainSearchResponse::getTlds()
{
	return tlds;
}

void
DomainSearchResponse::setTlds(std::list<std::string> tlds)
{
	this->tlds = tlds;
}

File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/DomainSearchResponse.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Lookup and suggestion results returned for a domain search.
 */
@Schema(description = "Lookup and suggestion results returned for a domain search.")
@Validated
@NotUndefined



public class DomainSearchResponse   {
  @JsonProperty("success")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private Boolean success = null;

  @JsonProperty("response_text")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String responseText = null;

  @JsonProperty("response_time")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String responseTime = null;

  @JsonProperty("lookup")
  @Valid
  private List<Object> lookup = null;
  @JsonProperty("suggest")
  @Valid
  private List<Object> suggest = null;
  @JsonProperty("tlds")
  @Valid
  private List<String> tlds = null;

  public DomainSearchResponse success(Boolean success) { 

    this.success = success;
    return this;
  }

  /**
   * Indicates whether the registrar search succeeded.
   * @return success
   **/
  
  @Schema(description = "Indicates whether the registrar search succeeded.")
  
  public Boolean isSuccess() {  
    return success;
  }



  public void setSuccess(Boolean success) { 
    this.success = success;
  }

  public DomainSearchResponse responseText(String responseText) { 

    this.responseText = responseText;
    return this;
  }

  /**
   * Human-readable status text from the registrar.
   * @return responseText
   **/
  
  @Schema(description = "Human-readable status text from the registrar.")
  
  public String getResponseText() {  
    return responseText;
  }



  public void setResponseText(String responseText) { 
    this.responseText = responseText;
  }

  public DomainSearchResponse responseTime(String responseTime) { 

    this.responseTime = responseTime;
    return this;
  }

  /**
   * Response time as reported by the registrar.
   * @return responseTime
   **/
  
  @Schema(description = "Response time as reported by the registrar.")
  
  public String getResponseTime() {  
    return responseTime;
  }



  public void setResponseTime(String responseTime) { 
    this.responseTime = responseTime;
  }

  public DomainSearchResponse lookup(List<Object> lookup) { 

    this.lookup = lookup;
    return this;
  }

  public DomainSearchResponse addLookupItem(Object lookupItem) {
    if (this.lookup == null) {
      this.lookup = new ArrayList<Object>();
    }
    this.lookup.add(lookupItem);
    return this;
  }

  /**
   * Availability lookup results for queried domains.
   * @return lookup
   **/
  
  @Schema(description = "Availability lookup results for queried domains.")
  
  public List<Object> getLookup() {  
    return lookup;
  }



  public void setLookup(List<Object> lookup) { 
    this.lookup = lookup;
  }

  public DomainSearchResponse suggest(List<Object> suggest) { 

    this.suggest = suggest;
    return this;
  }

  public DomainSearchResponse addSuggestItem(Object suggestItem) {
    if (this.suggest == null) {
      this.suggest = new ArrayList<Object>();
    }
    this.suggest.add(suggestItem);
    return this;
  }

  /**
   * Suggested alternative domains and availability data.
   * @return suggest
   **/
  
  @Schema(description = "Suggested alternative domains and availability data.")
  
  public List<Object> getSuggest() {  
    return suggest;
  }



  public void setSuggest(List<Object> suggest) { 
    this.suggest = suggest;
  }

  public DomainSearchResponse tlds(List<String> tlds) { 

    this.tlds = tlds;
    return this;
  }

  public DomainSearchResponse addTldsItem(String tldsItem) {
    if (this.tlds == null) {
      this.tlds = new ArrayList<String>();
    }
    this.tlds.add(tldsItem);
    return this;
  }

  /**
   * TLDs evaluated during the search.
   * @return tlds
   **/
  
  @Schema(description = "TLDs evaluated during the search.")
  
  public List<String> getTlds() {  
    return tlds;
  }



  public void setTlds(List<String> tlds) { 
    this.tlds = tlds;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DomainSearchResponse domainSearchResponse = (DomainSearchResponse) o;
    return Objects.equals(this.success, domainSearchResponse.success) &&
        Objects.equals(this.responseText, domainSearchResponse.responseText) &&
        Objects.equals(this.responseTime, domainSearchResponse.responseTime) &&
        Objects.equals(this.lookup, domainSearchResponse.lookup) &&
        Objects.equals(this.suggest, domainSearchResponse.suggest) &&
        Objects.equals(this.tlds, domainSearchResponse.tlds);
  }

  @Override
  public int hashCode() {
    return Objects.hash(success, responseText, responseTime, lookup, suggest, tlds);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DomainSearchResponse {\n");
    
    sb.append("    success: ").append(toIndentedString(success)).append("\n");
    sb.append("    responseText: ").append(toIndentedString(responseText)).append("\n");
    sb.append("    responseTime: ").append(toIndentedString(responseTime)).append("\n");
    sb.append("    lookup: ").append(toIndentedString(lookup)).append("\n");
    sb.append("    suggest: ").append(toIndentedString(suggest)).append("\n");
    sb.append("    tlds: ").append(toIndentedString(tlds)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
